core/src/test/scala/org/apache/spark/sql/delta/TimestampLocalFileSystem.scala

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright (2021) The Delta Lake Project Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.sql.delta
+
+import java.net.URI
+
+import org.apache.hadoop.conf.Configuration
+import org.apache.hadoop.fs.{DelegateToFileSystem, Path, RawLocalFileSystem}
+import org.apache.hadoop.fs.FileStatus
+
+/**
+ * This custom fs implementation is used for testing the msync calling in HDFSLogStore writes
```

Comment:
nit: add a simple comment to explain how it works
```
If its `msync` is not called, `listStatus` will return a stale result.
```

Reply:
will add